src/main/java/Main/Main.java

```diff
@@ -0,0 +1,39 @@
+package Main;
+
+import Controller.Controller;
+import Model.Board;
+import Model.FEN;
+import Observer.Observer;
+import View.MainView;
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import static java.lang.Thread.sleep;
+
+public class Main extends Application {
+
+    @Override
+    public void start(Stage stage) throws IOException, InterruptedException {
+
+
+
+        Observer ObserverMainView = new MainView(stage); // Start up View
+
+
+        Board board = new Board(ObserverMainView); // Start up Model
```

Comment:
this in an architectural leak. while it's definitely the right idea for the `Board` (model) to not hold a reference to the `MainView` (view) directly and instead store a reference to an `Observer`, this does still however mean that the model is communicating directly with the view.

There is nothing really _wrong_ with the view and model communicating, if that is your architectural approach. But it means in this app, all three components are communicating freely to each other.

In my opinion, the controller should be observing the model, and telling the view to update itself, and vice versa.

Reply:
![image](https://user-images.githubusercontent.com/67894560/153798810-d51a6df4-7681-48c0-8f22-a12db5a65b1c.png)

Reply:
yeh awesome this is exactly what you have, so if this is what your going for then its perfect.

I think what I am talking about is closer to MVP. All options work, just as long as its what you intended 👍